src/widget/wwidgetstack.cpp

```diff
@@ -74,24 +75,59 @@ QSize WWidgetStack::minimumSizeHint() const {
     return pWidget ? pWidget->minimumSizeHint() : QSize();
 }
 
+void WWidgetStack::showIndex(int index) {
+    if (m_bRespondToChanges) {
+        setCurrentIndex(index);
+    }
+}
+
 void WWidgetStack::hideIndex(int index) {
+    if (!m_bRespondToChanges) {
+        return;
+    }
     if (currentIndex() == index) {
         QMap<int, int>::const_iterator it = m_hideMap.find(index);
         if (it != m_hideMap.end()) {
             setCurrentIndex(*it);
         } else {
+            // TODO: This default behavior is a little odd, is it really what
+            // we want?  Or should we save the previously-selected page and then
+            // switch to that.
             setCurrentIndex((index + 1) % count());
         }
     }
 }
 
+void WWidgetStack::showEvent(QShowEvent*) {
+    int index = static_cast<int>(m_currentPageControl.get());
+
+    // Set the page triggers to match the current index.
+    for (QMap<int, ControlObject*>::iterator it = m_triggers.begin();
+            it != m_triggers.end(); ++it) {
+        it.value()->set(it.key() == index ? 1.0 : 0.0);
+    }
+
+    m_bRespondToChanges = true;
```

Comment:
respond-to-changes exists so that two widgetstacks can share the same trigger CO (show_samplers) and not get in to an argument about who is in charge.  It is a proxy for isvisible.  With the COThread change I can just say "isvisible"